### PR TITLE
Multi stream ios ut for new btree.

### PR DESCRIPTION
**Pull request by shashank-parulekar**

### CODE REVIEW

File: btree/btree.c [hunk 1]
```diff
@@ -3892,24 +3931,193 @@ static void m0_btree_ut_basic_kv_operations(void)
...
+		struct cb_data       put_data;
+		struct m0_btree_rec  rec;
+		m0_bcount_t          ksize  = sizeof key;
+		m0_bcount_t          vsize  = sizeof value;
+		void                *k_ptr  = &key;
+		void                *v_ptr  = &value;
+		uint32_t             stream_num;
+
+		for (stream_num = 0; stream_num < stream_count; stream_num++) {
+			int k;
```
Comment (RadhaGulhane13) [inline btree/btree.c:4029]: can you shift line 4018 to 4025 at line 2029 as new record for each put operation is required, as record can get updated by put operation, so better to use new record for each put operation

Comment (shashank-parulekar) [reply btree/btree.c:4029]: A new record is always sent to the put routine. I am sorry I donot follow your comments.

**Review by RadhaGulhane13 — APPROVED** [review]

File: btree/btree.c [hunk 2]
```diff
@@ -3695,12 +3702,12 @@ static void m0_btree_ut_basic_tree_operations(void)
...
-	M0_BTREE_OP_SYNC_WITH(&b_op.bo_op, m0_btree_destroy(btree, &b_op), &G,
-			      &b_op.bo_op_exec);
+	M0_BTREE_OP_SYNC_WITH_RC(&b_op.bo_op, m0_btree_destroy(btree, &b_op),
+				 &G, &b_op.bo_op_exec);
 
 	/** Create a new btree */
 	temp_node = m0_alloc_aligned((1024 + sizeof(struct nd)), 10);
-	M0_BTREE_OP_SYNC_WITH(&b_op.bo_op,
+	M0_BTREE_OP_SYNC_WITH_RC(&b_op.bo_op,
 			      m0_btree_create(temp_node, 1024, &btree_type, nt,
```
Comment (ydb242) [inline btree/btree.c:3711]: Alignment.

File: btree/btree.c [hunk 3]
```diff
@@ -3822,7 +3861,7 @@ static void m0_btree_ut_basic_kv_operations(void)
 
 	/** Create temp node space and use it as root node for btree */
 	temp_node = m0_alloc_aligned((1024 + sizeof(struct nd)), 10);
-	M0_BTREE_OP_SYNC_WITH(&b_op.bo_op,
+	M0_BTREE_OP_SYNC_WITH_RC(&b_op.bo_op,
 			      m0_btree_create(temp_node, 1024, &btree_type, nt,
```
Comment (ydb242) [inline btree/btree.c:3865]: Alignment.

File: btree/btree.c [hunk 4]
```diff
@@ -3857,7 +3896,7 @@ static void m0_btree_ut_basic_kv_operations(void)
 		ut_cb.c_act        = btree_kv_put_cb;
 		ut_cb.c_datum      = &put_data;
 
-		M0_BTREE_OP_SYNC_WITH(&kv_op.bo_op,
+		M0_BTREE_OP_SYNC_WITH_RC(&kv_op.bo_op,
 				      m0_btree_put(b_op.bo_arbor, tx, &rec,
```
Comment (ydb242) [inline btree/btree.c:3900]: Alignment.

File: btree/btree.c [hunk 5]
```diff
@@ -3892,24 +3931,193 @@ static void m0_btree_ut_basic_kv_operations(void)
 		search_key_in_tree.k_data =
 			M0_BUFVEC_INIT_BUF(&search_key_ptr, &search_key_size);
 
-		M0_BTREE_OP_SYNC_WITH(&kv_op.bo_op,
+		M0_BTREE_OP_SYNC_WITH_RC(&kv_op.bo_op,
 				      m0_btree_get(b_op.bo_arbor,
```
Comment (ydb242) [inline btree/btree.c:3935]: Alignment

File: btree/btree.c [hunk 6]
```diff
@@ -3892,24 +3931,193 @@ static void m0_btree_ut_basic_kv_operations(void)
...
 						   &search_key_in_tree,
 						   &ut_cb, 0, &kv_op), &G,
-				      &b_op.bo_op_exec);
+						   &b_op.bo_op_exec);
 
 		for (i = 1; i < 2048; i++) {
 			search_key = key;
-			M0_BTREE_OP_SYNC_WITH(&kv_op.bo_op,
+			M0_BTREE_OP_SYNC_WITH_RC(&kv_op.bo_op,
 					      m0_btree_nxt(b_op.bo_arbor,
```
Comment (ydb242) [inline btree/btree.c:3943]: Alignment

File: btree/btree.c [hunk 7]
```diff
@@ -3892,24 +3931,193 @@ static void m0_btree_ut_basic_kv_operations(void)
...
-					      &G, &b_op.bo_op_exec);
+							   &G,
+							   &b_op.bo_op_exec);
+		}
+	}
+
+	m0_btree_close(b_op.bo_arbor);
+	M0_BTREE_OP_SYNC_WITH_RC(&b_op.bo_op,
+			      m0_btree_destroy(b_op.bo_arbor, &b_op), &G,
+			      &b_op.bo_op_exec);
```
Comment (ydb242) [inline btree/btree.c:3954]: Alignment for above 2 lines

File: btree/btree.c [hunk 8]
```diff
@@ -3892,24 +3931,193 @@ static void m0_btree_ut_basic_kv_operations(void)
...
+	 *  3) Confirms the records are present in the tree.
+	 *  4) Deletes all the records from the tree using multiple streams.
+	 *  5) Close the btree
+	 *  6) Destroy the btree
+	 */
+
+	/** Create temp node space and use it as root node for btree */
+	temp_node = m0_alloc_aligned((1024 + sizeof(struct nd)), 10);
+	M0_BTREE_OP_SYNC_WITH_RC(&b_op.bo_op,
+			      m0_btree_create(temp_node, 1024, &btree_type, nt,
```
Comment (ydb242) [inline btree/btree.c:4014]: Alignment.

File: btree/btree.c [hunk 9]
```diff
@@ -3892,24 +3931,193 @@ static void m0_btree_ut_basic_kv_operations(void)
...
+
+		get_data.key         = &get_key;
+		get_data.value       = &get_value;
+		get_data.check_value = true;
+
+		ut_cb.c_act   = btree_kv_get_cb;
+		ut_cb.c_datum = &get_data;
+
+		M0_BTREE_OP_SYNC_WITH_RC(&kv_op.bo_op,
+				      m0_btree_get(b_op.bo_arbor,
```
Comment (ydb242) [inline btree/btree.c:4083]: Alignment

File: btree/btree.c [hunk 10]
```diff
@@ -3892,24 +3931,193 @@ static void m0_btree_ut_basic_kv_operations(void)
...
+			rec.r_val          = M0_BUFVEC_INIT_BUF(&v_ptr, &vsize);
+
+			put_data.key       = &rec.r_key;
+			put_data.value     = &rec.r_val;
+
+			ut_cb.c_act        = btree_kv_put_cb;
+			ut_cb.c_datum      = &put_data;
+
+			M0_BTREE_OP_SYNC_WITH_RC(&kv_op.bo_op,
+					      m0_btree_put(b_op.bo_arbor, tx,
```
Comment (ydb242) [inline btree/btree.c:4046]: Alignment..

File: btree/btree.c [hunk 11]
```diff
@@ -3892,24 +3931,193 @@ static void m0_btree_ut_basic_kv_operations(void)
...
+			M0_BUFVEC_INIT_BUF(&delete_key_ptr, &delete_key_size);
+
+		ut_cb.c_act   = btree_kv_del_cb;
+		ut_cb.c_datum = &delete_data;
+
+		for (stream_num = 0; stream_num < stream_count; stream_num++) {
+			delete_key = i + (stream_num * recs_per_stream);
+
+			M0_BTREE_OP_SYNC_WITH_RC(&kv_op.bo_op,
+					      m0_btree_del(b_op.bo_arbor,
```
Comment (ydb242) [inline btree/btree.c:4112]: Alignment.

File: btree/btree.c [hunk 12]
```diff
@@ -3892,24 +3931,193 @@ static void m0_btree_ut_basic_kv_operations(void)
...
+							   &delete_key_in_tree,
+							   &ut_cb, 0, &kv_op),
+						 &G, &b_op.bo_op_exec);
 		}
 	}
 
 	m0_btree_close(b_op.bo_arbor);
-	M0_BTREE_OP_SYNC_WITH(&b_op.bo_op,
+	M0_BTREE_OP_SYNC_WITH_RC(&b_op.bo_op,
 			      m0_btree_destroy(b_op.bo_arbor, &b_op), &G,
```
Comment (ydb242) [inline btree/btree.c:4121]: Alignment.

**Review by ydb242 — APPROVED** [review]
Looks good to me!

**Pull request merged**